Systematic index out of range panic on a specific Dockerfile

Hi there,

Trying out buildkit for the first time, after a few successful attempts at building dockerfiles I have a panic on one specific Dockerfile (which I can't show unfortunately).

Might be my mistake as those are my first steps with buildkit but it might be worth sharing.

Running `docker 18.06.1-ce`, Arch Linux (`4.18.10 kernel`), 

- *buildkit version* (pulled from master, being adventurous)
```
buildctl github.com/moby/buildkit 0.0.0 5d8be2c4c033637a1613bdc56c0461ce6613c1fc
```

- *buildctl* logs:
```
$ sudo buildctl build --frontend dockerfile.v0 --local context=. --local dockerfile=.
[+] Building 1.4s (3/3) FINISHED                                                                           
 => [internal] load .dockerignore                                                                     0.0s
 => => transferring context: 34B                                                                      0.0s
 => [internal] load build definition from Dockerfile                                                  0.0s
 => => transferring dockerfile: 32B                                                                   0.0s
 => [internal] load metadata for docker.io/library/ubuntu:latest                                      1.3s
error: failed to solve: rpc error: code = Unavailable desc = transport is closing
```

- *buildkitd* logs [panic]:
```
$ sudo buildkitd --debug --root /var/lib/buildkit
DEBU[0000] auto snapshotter: using overlayfs            
INFO[0000] found worker "blbl847r8gyac3tcpfdiv1et1", labels=map[org.mobyproject.buildkit.worker.executor:oci org.mobyproject.buildkit.worker.snapshotter:overlayfs org.mobyproject.buildkit.worker.hostname:kxcw], platforms=[linux/amd64] 
WARN[0000] skipping containerd worker, as "/run/containerd/containerd.sock" does not exist 
INFO[0000] found 1 workers, default="blbl847r8gyac3tcpfdiv1et1" 
WARN[0000] currently, only the default worker can be used. 
INFO[0000] running server on /run/buildkit/buildkitd.sock 
DEBU[0009] session started                              
DEBU[0009] reusing ref for local: r8gt1uaur40906i5xsd4yuidj 
DEBU[0009] reusing ref for local: vign0bhlkev0787xzzk1rc3o7 
DEBU[0009] diffcopy took: 2.26169ms                     
DEBU[0009] diffcopy took: 1.587741ms                    
DEBU[0009] saved r8gt1uaur40906i5xsd4yuidj as local.sharedKey:dockerfile:Dockerfile: 
DEBU[0009] saved vign0bhlkev0787xzzk1rc3o7 as local.sharedKey:context:.dockerignore: 
DEBU[0009] resolving                                    
DEBU[0009] do request                                    request.headers=map[Accept:[application/vnd.docker.distribution.manifest.v2+json, application/vnd.docker.distribution.manifest.list.v2+json, application/vnd.oci.image.manifest.v1+json, application/vnd.oci.image.index.v1+json, *]] request.method=HEAD url="https://registry-1.docker.io/v2/library/ubuntu/manifests/latest"
DEBU[0010] fetch response received                       response.headers=map[Strict-Transport-Security:[max-age=31536000] Content-Type:[application/json; charset=utf-8] Docker-Distribution-Api-Version:[registry/2.0] Www-Authenticate:[Bearer realm="https://auth.docker.io/token",service="registry.docker.io",scope="repository:library/ubuntu:pull"] Date:[Thu, 11 Oct 2018 17:04:25 GMT] Content-Length:[157]] status="401 Unauthorized" url="https://registry-1.docker.io/v2/library/ubuntu/manifests/latest"
DEBU[0010] Unauthorized                                  header="Bearer realm="https://auth.docker.io/token",service="registry.docker.io",scope="repository:library/ubuntu:pull""
DEBU[0010] do request                                    request.headers=map[Accept:[application/vnd.docker.distribution.manifest.v2+json, application/vnd.docker.distribution.manifest.list.v2+json, application/vnd.oci.image.manifest.v1+json, application/vnd.oci.image.index.v1+json, *]] request.method=HEAD url="https://registry-1.docker.io/v2/library/ubuntu/manifests/latest"
DEBU[0011] fetch response received                       response.headers=map[Content-Length:[2035] Content-Type:[application/vnd.docker.distribution.manifest.list.v2+json] Docker-Content-Digest:[sha256:de774a3145f7ca4f0bd144c7d4ffb2931e06634f11529653b23eba85aef8e378] Docker-Distribution-Api-Version:[registry/2.0] Etag:["sha256:de774a3145f7ca4f0bd144c7d4ffb2931e06634f11529653b23eba85aef8e378"] Date:[Thu, 11 Oct 2018 17:04:26 GMT] Strict-Transport-Security:[max-age=31536000]] status="200 OK" url="https://registry-1.docker.io/v2/library/ubuntu/manifests/latest"
DEBU[0011] resolved                                      desc.digest=sha256:de774a3145f7ca4f0bd144c7d4ffb2931e06634f11529653b23eba85aef8e378
DEBU[0011] fetch                                         digest=sha256:de774a3145f7ca4f0bd144c7d4ffb2931e06634f11529653b23eba85aef8e378 mediatype="application/vnd.docker.distribution.manifest.list.v2+json" size=2035
DEBU[0011] fetch                                         digest=sha256:a819482773d99bbbb570626b6101fa37cd93a678581ee564e89feae903c95f20 mediatype="application/vnd.docker.distribution.manifest.v2+json" size=1357
DEBU[0011] fetch                                         digest=sha256:cd6d8154f1e16e38493c3c2798977c5e142be5e5d41403ca89883840c6d51762 mediatype="application/vnd.docker.container.image.v1+json" size=3615
panic: runtime error: index out of range

goroutine 231 [running]:
github.com/moby/buildkit/frontend/dockerfile/dockerfile2llb.dispatchCopy(0xc000446c00, 0xc000563600, 0x2, 0x2, 0xf8c2c0, 0xc00056ec30, 0xf97e80, 0xc0004543f0, 0x0, 0x0, ...)
	/go/src/github.com/moby/buildkit/frontend/dockerfile/dockerfile2llb/convert.go:656 +0x26de
github.com/moby/buildkit/frontend/dockerfile/dockerfile2llb.dispatch(0xc000446c00, 0xf88880, 0xc000282600, 0x0, 0x0, 0x0, 0xc000563660, 0xc000224e40, 0x8, 0x8, ...)
	/go/src/github.com/moby/buildkit/frontend/dockerfile/dockerfile2llb/convert.go:500 +0x712
github.com/moby/buildkit/frontend/dockerfile/dockerfile2llb.Dockerfile2LLB(0xf97dc0, 0xc000317a40, 0xc0000e8400, 0x1cc, 0x3cc, 0x0, 0x0, 0x7fcb74f32bd8, 0xc00022a310, 0xc000584630, ...)
	/go/src/github.com/moby/buildkit/frontend/dockerfile/dockerfile2llb/convert.go:340 +0x1a94
github.com/moby/buildkit/frontend/dockerfile/builder.Build.func3.1(0x8, 0xed2d00)
	/go/src/github.com/moby/buildkit/frontend/dockerfile/builder/build.go:259 +0x44e
github.com/moby/buildkit/vendor/golang.org/x/sync/errgroup.(*Group).Go.func1(0xc000584600, 0xc000224d80)
	/go/src/github.com/moby/buildkit/vendor/golang.org/x/sync/errgroup/errgroup.go:58 +0x57
created by github.com/moby/buildkit/vendor/golang.org/x/sync/errgroup.(*Group).Go
	/go/src/github.com/moby/buildkit/vendor/golang.org/x/sync/errgroup/errgroup.go:55 +0x66
```

--- a/frontend/dockerfile/dockerfile2llb/convert.go
+++ b/frontend/dockerfile/dockerfile2llb/convert.go
@@ -653,7 +653,7 @@ func dispatchCopy(d *dispatchState, c instructions.SourcesAndDest, sourceState l
 	img := llb.Image(opt.copyImage, llb.MarkImageInternal, llb.Platform(opt.buildPlatforms[0]), WithInternalName("helper image for file operations"))
 
 	dest := path.Join(".", pathRelativeToWorkingDir(d.state, c.Dest()))
-	if c.Dest() == "." || c.Dest()[len(c.Dest())-1] == filepath.Separator {
+	if c.Dest() == "." || c.Dest() == "" || c.Dest()[len(c.Dest())-1] == filepath.Separator {
 		dest += string(filepath.Separator)
 	}
 	args := []string{"copy"}

